src/libmongoc/src/mongoc/mongoc-write-command.c

```diff
@@ -698,23 +699,20 @@ _mongoc_write_opmsg (mongoc_write_command_t *command,
    }
 
    do {
-      uint32_t ulen;
-      memcpy (&ulen, command->payload.data + payload_batch_size + payload_total_offset, 4);
-      ulen = BSON_UINT32_FROM_LE (ulen);
+      const int32_t len = mlib_read_u32le (command->payload.data + payload_batch_size + payload_total_offset);
```

Comment:
This change produces:
```
warning: conversion to ‘int32_t’ {aka ‘const int’} from ‘uint32_t’ {aka ‘unsigned int’} may change the sign of the result [-Wsign-conversion]
```
Consider reverting/revising these changes to preserve the assertion of `ulen` before the cast to signed `len` to avoid implementation-defined behavior for unsigned->signed conversion.